attestation_object.go

```diff
@@ -0,0 +1,40 @@
+package webauthn
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/fxamacker/cbor/v2"
+)
+
+// ErrInvalidAttestationObject indicates an invalid attestation object.
+var ErrInvalidAttestationObject = errors.New("invalid attestation object")
+
+// An AttestationObject conveys authenticator data and an attestation statement.
+type AttestationObject struct {
+	// The AuthData encodes contextual bindings made by the authenticator. These bindings are
+	// controlled by the authenticator itself, and derive their trust from the WebAuthn Relying Party's assessment
+	// of the security properties of the authenticator.
+	AuthData []byte `json:"authData"`
+	// The Format is the manner in which the signature is represented and the various contextual bindings are
+	// incorporated into the attestation statement by the authenticator.
+	Format string `json:"fmt"`
+	// The Statement is a signed data object, containing statements about a public key credential itself and
+	// the authenticator that created it.
+	Statement map[string]interface{} `json:"attStmt,omitempty"`
+}
+
+// UnmarshalAttestationObject unmarshals an attestation object from a slice of bytes. It expects a CBOR-encoded
+// map.
+func UnmarshalAttestationObject(raw []byte) (attestationObject *AttestationObject, remaining []byte, err error) {
```

Comment:
We should probably add a `io.LimitReader` to guard against _very_ large / indefinite length paylods. I'm not sure of the best place to do it.

https://github.com/fxamacker/cbor#limitations

Reply:
It accepts a slice of bytes not a reader. What's the value of using a limit reader when it's already limited to the size of the slice of bytes?

Reply:
Ah, fair point.